### PR TITLE
feat(contactsmenu): Show user status

### PR DESCRIPTION
**Pull request by ChristophWurst**
* Resolves part 1 of https://github.com/nextcloud/server/issues/40559

## Summary

Show the user status for system contacts if they have a status set. Show the email address otherwise.

| Before | After |
|--------|--------|
| Bildschirmfoto vom 2023-10-17 13-18-45 | Bildschirmfoto vom 2023-10-18 20-33-12 | 

## TODO

- [x] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
  - https://github.com/nextcloud/documentation/pull/11226
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by ChristophWurst** [comment]
requesting early design review based on the before/after screenshots

I think the feature is done. I only have to adjust/write tests

**Review by nimishavijay — COMMENTED** [review]
Looks good! small cosmetic changes:

- [x] "Show all contacts" remove ellipsis and text in bold so it looks like a button 
- [x] size of avatar could be 44px (ignore if already the case, it looks smaller than 44px to me)
- [x] increase space between avatar and text by 2px 
- [x] decrease space between main text and subline by 2px
Basically similar to the styling of a list item in, for eg. the contacts app.
image
- [ ] Also, if clicking on the item opens the profile anyway, I'd say we don't need a separate profile button, it could just be a 3 dot menu

**Comment by nimishavijay** [comment]
> Do you want me to remove the profile action or just move it into the actions menu? 

Ah okay, I understood, so for emails that would be a mail action? 

How about we make Talk the priority action and keep profile in the 3 dot menu. That way, when you open the contact search you see users whom you can Talk to and email addresses that you can send mails to, and if you click on the user you would go to their profile anyway. 

My idea is that you are probably looking for a contact to do something specific like chat or email more than view the profile. What do you think?

**Review by jancborchardt — APPROVED** [review]
Looks nice! :) I’d agree with @nimishavijay that having "Talk" as the primary action makes more sense as it’s a more direct action.
However then clicking the name/status part of the person should do the same to not have 4 different click areas (avatar, name, action icon, action menu), and "Profile" has to be put into the action menu – so your call if this should be in here or in a separate pull request @ChristophWurst.

**Comment by ChristophWurst** [comment]
Let's do these changes separately :+1:

File: lib/private/Contacts/ContactsMenu/ActionProviderStore.php [hunk 1]
```diff
@@ -47,7 +48,6 @@ public function __construct(
 	}
 
 	/**
-	 * @return IProvider[]
```
Comment (nickvergessen) [inline lib/private/Contacts/ContactsMenu/ActionProviderStore.php:50]:
```
	 * @return IProvider[]|IBulkProvider[]
```

File: lib/private/Contacts/ContactsMenu/Entry.php [hunk 2]
```diff
@@ -142,7 +159,7 @@ public function getProperty(string $key): mixed {
 	}
 
 	/**
-	 * @return array{id: int|string|null, fullName: string, avatar: string|null, topAction: mixed, actions: array, lastMessage: '', emailAddresses: string[], profileTitle: string|null, profileUrl: string|null}
+	 * @return array{id: int|string|null, fullName: string, avatar: string|null, topAction: mixed, actions: array, lastMessage: '', emailAddresses: string[], profileTitle: string|null, profileUrl: string|null, status: string|null, statusMessage: null|string, statusIcon: null|string, isUser: bool, uid: mixed}
```
Comment (nickvergessen) [inline lib/private/Contacts/ContactsMenu/Entry.php:162]: I guess you need to recreate openapi docs?

**Review by nickvergessen — APPROVED** [review]
Only checked PHP

File: lib/private/Contacts/ContactsMenu/Entry.php (hunk 2 above)
Comment (ChristophWurst) [reply lib/private/Contacts/ContactsMenu/Entry.php:162]: > OpenAPI specifications up to date. Carry on

File: lib/private/Contacts/ContactsMenu/ActionProviderStore.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/private/Contacts/ContactsMenu/ActionProviderStore.php:50]: https://psalm.dev/r/705c8b2f87 vs https://psalm.dev/r/eef0455894 

:smiling_face_with_tear:

Comment (nickvergessen) [reply lib/private/Contacts/ContactsMenu/ActionProviderStore.php:50]: Oh come on psalm :(

Comment (nickvergessen) [reply lib/private/Contacts/ContactsMenu/ActionProviderStore.php:50]: Hmm could call the method bulkProcess in the bulk provider, then a provider could be both :D
But well the else if is already taking care. So yolo from here

**Review by marcoambrosini — APPROVED** [review]

**Comment by ChristophWurst** [comment]
Red CI unrelated. Cypress is random and postgres timed out.

**Comment by ChristophWurst** [comment]
I hovered the button to stop automerge and when I clicked the page jumped and my click went to the upate button. Great. Just four more hours.

**Pull request merged**